File: mywebsite-parent-v2.0/mywebsite-protal-web/src/main/webapp/angularjs/js/service/blogCategoryService.js
app.service('blogCategoryService', function ($http) {


    /**
     * 查询所有分类
     * @returns {*}
     */
    this.findAll = function () {
        return $http.get('../blogCategory/findAll');
    };

    /**
     * 通过分类id查询
     * @param id
     * @returns {*}
     */
    this.findOneById = function (id) {
        return $http.get('../blogCategory/findOneById?id=' + id);
    };

    /**
     * 通过分类名查询
     * @param name
     * @returns {*}
     */
    this.findOneByName = function (name) {
        return $http.get('../blogCategory/findOneByName?name=' + name);
    };

    /**
     * 添加分类
     * @param tableEntity
     * @returns {*}
     */
    this.add = function (tableEntity) {
        return $http.post('../blogCategory/add', tableEntity);
    };

    /**
     * 更新分类
     * @param tableEntity
     * @returns {*}
     */
    this.update = function (tableEntity) {
        return $http.post('../blogCategory/update', tableEntity);
    };

    /**
     * 删除分类
     * @param ids
     * @returns {*}
     */
    this.deleteIds = function (ids) {
        return $http.get('../blogCategory/deleteIds?ids=' + ids);
    };

});

File: mywebsite-parent-v2.0/mywebsite-manager-web/src/main/java/xyz/lirui123/mywebsite/manager/service/BlogCategoryService.java
package xyz.lirui123.mywebsite.manager.service;


import org.springframework.stereotype.Component;
import xyz.lirui123.mywebsite.pojo.TbBlogCategory;
import xyz.lirui123.mywebsite.response.ResponseResult;

public interface BlogCategoryService {

    /**
     * 查询所有分类
     * @return
     */
    public abstract ResponseResult findAll();

    /**
     * 根据id查询分类
     * @param id
     * @return
     */
    public abstract TbBlogCategory findOneById(Long id);

    /**
     * 根据分类名称查询
     * @param name
     * @return
     */
    public abstract TbBlogCategory findOneByName(String name);

    /**
     * 添加分类
     * @param blogCategory
     * @return
     */
    public abstract ResponseResult add(TbBlogCategory blogCategory);

    /**
     * 更新分类
     * @param blogCategory
     * @return
     */
    public abstract ResponseResult update(TbBlogCategory blogCategory);

    /**
     * 删除分类
     * @param ids
     * @return
     */
    public abstract ResponseResult deleteIds(Long[] ids);
}
